### PR TITLE
feat(documentation): add visual regression test for textarea component

### PR DESCRIPTION
**Pull request by davidritter-dotcom**

### CODE REVIEW

File: packages/documentation/cypress/snapshots/components/textarea.snapshot.ts
Comment (gfellerph) [inline packages/documentation/cypress/snapshots/components/textarea.snapshot.ts:1]: This file should actually contain the test code.

File: packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                .filter(
+                  (args: Args) => !(args.hiddenLabel === true && args.floatingLabel === false),
+                )
+                .filter(
+                  (args: Args) => !(args.hint === longText && args.validation === 'is-invalid'),
+                )
+                .filter(
+                  (args: Args) => !(args.label === longText && args.validation === 'is-invalid'),
+                )
+                .map((args: Args) => {
```
Comment (gfellerph) [inline packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts:42]: Reduce duplicates even further, there are more than 10 examples of the valid textarea.

image

**Review by gfellerph — CHANGES_REQUESTED** [review]

File: packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts
Comment (oliverschuerch) [inline packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts:1]: Take into consideration reusing the snapshot test file for select: https://next.design-system.post.ch/iframe.html?id=snapshots--select

**Review by oliverschuerch — CHANGES_REQUESTED** [review]
Again, I recommend using the [select.snapshot.stories.ts](https://github.com/swisspost/design-system/blob/main/packages/documentation/src/stories/components/select/select.snapshot.stories.ts) file as a starting point and change whatever is necessary.

And to save a little bit of vertical space, we could add the attribute `rows` with a value of `1` or `2` to every textarea and only add one example with its default height per size + floating-label.

**Comment by oliverschuerch** [comment]
@b1aserlu
Please merge the main branch into this branch, so we get a clean list of (in this PR) changed files only.

File: packages/documentation/src/stories/components/input/input.snapshot.stories.ts [hunk 2]
```diff
@@ -40,11 +40,13 @@ function renderInputSnapshot(_args: Args, context: StoryContext) {
                     <div>
                       ${args.title !== undefined && args.title
                         ? html`
+
                             <h4>
                               ${Object.entries(context.argTypes.size.control.labels)
                                 .filter(([key, value]) => key === args.size)
                                 .map(s => s[1])}
                             </h4>
+
```
Comment (oliverschuerch) [inline packages/documentation/src/stories/components/input/input.snapshot.stories.ts:49]: Remove unnecessary empty lines.

File: packages/documentation/src/stories/components/select/select.snapshot.stories.ts [hunk 3]
```diff
@@ -1,116 +1,80 @@
...
+                  !args.multipleSize ||
+                  (args.multipleSize && context.args.multiple === true),
+              )
+              .map(
+                (args: Args) =>
+                  html`
+                    <div>
+                      ${args.title !== undefined && args.title
+                        ? html`
+                            <h4>${args.size}</h4>
```
Comment (oliverschuerch) [inline packages/documentation/src/stories/components/select/select.snapshot.stories.ts:42]:
```suggestion
                            <h4>
                              ${Object.entries(context.argTypes.size.control.labels)
                                .filter(([key, value]) => key === args.size)
                                .map(s => s[1])}
                            </h4>
```

File: packages/documentation/src/stories/components/select/select.snapshot.stories.ts [hunk 4]
```diff
@@ -1,116 +1,80 @@
...
-      ...bombArgs({
-        size: context.argTypes.size.options,
-        disabled: [false, true],
-        validation: context.argTypes.validation.options,
-      }),
-    ]
-      // remove disabled & validated examples
-      .filter((args: Args) => !(args.disabled && args.validation !== 'null'));
+function renderSelectSnapshot(_args: Args, context: StoryContext) {
+  const combinations = [
```
Comment (oliverschuerch) [inline packages/documentation/src/stories/components/select/select.snapshot.stories.ts:13]: There is no example with a select option (like a filled input).

File: packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export default {
+  ...meta,
+  title: 'Snapshots',
+};
+
+type Story = StoryObj;
+
+export const Textarea: Story = {
+  render: (_args: Args, context: StoryContext) => {
+    const combinations = [
```
Comment (oliverschuerch) [inline packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts:15]: There is no example for a filled textarea.

File: packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            <div class="${bg} d-flex gap-3 flex-column p-3">
+              <h3>Sizes</h3>
+              ${getCombinations('size', context.argTypes.size.options, combinations)
+                .map(
+                  (args: Args) =>
+                    html`
+                      <div>
+                        ${args.title !== undefined && args.title
+                          ? html`
+                              <h4>${args.size}</h4>
```
Comment (oliverschuerch) [inline packages/documentation/src/stories/components/textarea/textarea.snapshot.stories.ts:40]:
```suggestion
                              <h4>
                              ${Object.entries(context.argTypes.size.control.labels)
                                .filter(([key, value]) => key === args.size)
                                .map(s => s[1])}
                            </h4>
```

File: packages/documentation/src/utils/inputComponentsGetCombinations.ts
Comment (oliverschuerch) [inline packages/documentation/src/utils/inputComponentsGetCombinations.ts:1]: Remove unnecessary empty lines.

**Review by oliverschuerch — CHANGES_REQUESTED** [review]

**Comment by b1aserlu** [comment]
In the PR #1165 there was a problem that the snapshot test turned out to be blank I tried to fix this with the PR #2095 . As this PR handles a similar thing and will likely run in the same problem check out the changes I made and implement something similar to prevent this from happening,

**Review by gfellerph — APPROVED** [review]

File: packages/documentation/cypress/snapshots/components/input.snapshot.ts [hunk 7]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (oliverschuerch) [inline packages/documentation/cypress/snapshots/components/input.snapshot.ts:1]:
```suggestion
```

File: packages/documentation/src/stories/components/select/select.snapshot.stories.ts (hunk 4 above)
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/select/select.snapshot.stories.ts:13]: Now, every example has a selected value.

**Review by oliverschuerch — CHANGES_REQUESTED** [review]

**Review by oliverschuerch — APPROVED** [review]

**Pull request merged**